pkg/gadget-service/service-oci.go

```diff
@@ -50,6 +52,17 @@ func (s *Service) GetGadgetInfo(ctx context.Context, req *api.GetGadgetInfoReque
 		return nil, fmt.Errorf("expected version to be %d, got %d", api.VersionGadgetInfo, req.Version)
 	}
 
+	p, ok := peer.FromContext(ctx)
+	if ok {
```

Comment:
```suggestion
	if ok && p.AuthInfo != nil {
```

Otherwise it fails when with `Error: fetching gadget information: getting gadget info: rpc error: code = Unknown desc = authentication information are not TLS: <nil>` when no TLS information is provided.

(same below)